src/translator/sentence_splitter.cpp

```diff
@@ -0,0 +1,53 @@
+#include "sentence_splitter.h"
+#include "common/cli_helper.h"
+#include "common/logging.h"
+#include "common/options.h"
+#include <string>
+
+namespace marian {
+namespace bergamot {
+
+SentenceSplitter::SentenceSplitter(marian::Ptr<marian::Options> options)
+    : options_(options) {
+
+  std::string smode_str = options_->get<std::string>("ssplit-mode", "");
+  mode_ = string2splitmode(smode_str);
+  std::string ssplit_prefix_file =
+      options_->get<std::string>("ssplit-prefix-file", "");
+
+  if (ssplit_prefix_file.size()) {
+    ssplit_prefix_file = marian::cli::interpolateEnvVars(ssplit_prefix_file);
+
+    LOG(info, "Loading protected prefixes for sentence splitting from {}",
+        ssplit_prefix_file);
+
+    ssplit_.load(ssplit_prefix_file);
+  } else {
+    LOG(warn, "Missing list of protected prefixes for sentence splitting. "
+              "Set with --ssplit-prefix-file.");
+  }
+}
+
+ug::ssplit::SentenceStream
+SentenceSplitter::createSentenceStream(const string_view &input) {
+  std::string_view input_converted(input.data(), input.size());
```

Comment:
Convert this to `pcrecpp::StringPiece` with a bunch of ifdef guards and CMake, is an easy hack which will hurt in the long run. 

Ideal for both of us is a switch to PCRE2. @abhi-agg.

Reply:
This place is the only bottleneck to merge this into wasm-integration (not integration), potentially. I've gotten builds succeeding till here, beyond this was too much work. I don't know an estiamte beyond this what errors will surface in builds. native is building, wasm fails with pcre2. 
@abhi-agg

Reply:
[Patch](https://gist.github.com/jerinphilip/64679f42fa6929a3d2c860b16bb6f55e) which does the PCRECPP hack. Not going to push this in, would rather you switch to PCRE2. WASM [builds successfully](http://ix.io/2PZ2). At least on my development machine.

Reply:
The problem with the way pcre / pcre2 is compiled here is that it does not seem to include ICU, which is what we need, because we need character properties for sentence splitting. I'd also rather not bundle pcre2 with ssplit-cpp, because in for native compilation one can use binary packages that are available for most distributions. In my oponion pcre2 with icu should be stand-alone projects so that they can be re-used elsewere. I've bee trying to compile icu with emscripten since the meeting yesterday without success.

Reply:
@ugermann Linking https://github.com/ugermann/ssplit-cpp/issues/6 to move conversation there.

Reply:
Builds work with the merge of this [PR](https://github.com/ugermann/ssplit-cpp/pull/10)